### PR TITLE
List all challenges without login

### LINKED ISSUE #1247: Challenge List page should not require login
What the title says. 

Just like the VQA challenge details are not being auth-wall, everything else should be visible as well.

### PR DESCRIPTION
**Pull request by xamfy**
fixes #1247 
image 1
image 2

### CODE REVIEW

**Comment by deshraj** [comment]
@xamfy can you please share the screenshot of the same?

**Comment by xamfy** [comment]
@deshraj sure

File: frontend/src/js/controllers/challengeListAllCtrl.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        .controller('ChallengeListAllCtrl', ChallengeListAllCtrl);
+
+    ChallengeListAllCtrl.$inject = ['utilities'];
+
+    function ChallengeListAllCtrl(utilities) {
+        var vm = this;
+
+        utilities.showLoader();
+
+        vm.imgUrlObj = {
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeListAllCtrl.js]: We are not using these images anymore. Can you please remove them.

File: frontend/src/js/controllers/challengeListAllCtrl.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+                utilities.hideLoader();
+            }
+        };
+
+        utilities.sendRequest(parameters);
+
+
+
+        // utilities.showLoader();
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/challengeListAllCtrl.js]: Please remove the comments.

File: frontend/src/js/route-config/route-config.js [hunk 3]
```diff
@@ -183,6 +183,15 @@
             authenticate: true
         };
 
+        var list_all_challenges = {
+            name: 'list-all-challenges',
```
Comment (RishabhJain2018) [inline frontend/src/js/route-config/route-config.js]: Please change the name to `all-challenges`

File: frontend/src/js/route-config/route-config.js [hunk 4]
```diff
@@ -183,6 +183,15 @@
             authenticate: true
         };
 
+        var list_all_challenges = {
```
Comment (RishabhJain2018) [inline frontend/src/js/route-config/route-config.js]: Please change the varibale name to `all_challenges`

File: frontend/src/js/route-config/route-config.js [hunk 5]
```diff
@@ -445,6 +454,7 @@
         // challenges list page
         $stateProvider.state(challenge_main);
         $stateProvider.state(challenge_list);
+        $stateProvider.state(list_all_challenges);
```
Comment (RishabhJain2018) [inline frontend/src/js/route-config/route-config.js]: Same as above.

File: frontend/src/views/web/challenge/list_all_challenges.html [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                    </div>
+                    <div class=" btn-card-detail waves-effect waves-dark w-300 fs-14"> <strong>View Details </strong> &nbsp; </div>
+                </div>
+            </a>
+        </div>
+    </div>
+    <br>
+</section>
+    <landing-footer></landing-footer>
+</div>
```
Comment (RishabhJain2018) [inline frontend/src/views/web/challenge/list_all_challenges.html]: Please leave a blank line.

File: frontend/src/views/web/landing.html [hunk 7]
```diff
@@ -47,7 +47,7 @@ <h4 class="text-dark-black"><strong>EvalAI</strong></h4>
                             <p>
                                 Explore other past, ongoing and upcoming challenges.
                             </p>
-                            <a class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-light fs-14 mt-12" ui-sref="web.challenge-main">View All</a>
+                            <a class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-light fs-14 mt-12" ui-sref="list-all-challenges">View All</a>
```
Comment (RishabhJain2018) [inline frontend/src/views/web/landing.html]: Please change `ui-sref` accordingly.

**Comment by RishabhJain2018** [comment]
@deshraj @aka-jain  I suggest that the details of the challenge should also be not auth-walled. What do you say?
Currently in this PR the challenge list is not-walled.

**Comment by deshraj** [comment]
Yeah yeah, both the challenge list and challenge details have to be public. @xamfy can you please add support for making the challenge details public as well? Thanks for your valuable contribution. :)

**Comment by xamfy** [comment]
@deshraj okay, will do it.

**Comment by aka-jain** [comment]
Any update on this? @xamfy

**Comment by xamfy** [comment]
@aka-jain Sorry, couldn't work due to tests. I'm doing it now.

**Comment by deshraj** [comment]
@xamfy: any update on this? Please let us know if you are facing any issue.

**Comment by xamfy** [comment]
@deshraj I'm currently preparing for tests. I've done the list part. Would you mind taking over the details part?

**Comment by deshraj** [comment]
@RishabhJain2018 / @isht3  would you mind working on this feature and get it done soon? Please let me know if any of you cannot work on this. I can then take over this.

**Comment by deshraj** [comment]
Thanks @xamfy for letting us know about your exams. All the best for your exams.

**Comment by xamfy** [comment]
Thanks @deshraj

**Comment by RishabhJain2018** [comment]
@deshraj I will take it up. :)

**Comment by RishabhJain2018** [comment]
@deshraj @aka-jain I am taking this up.

**Comment by aka-jain** [comment]
Sure @RishabhJain2018

**Comment by aka-jain** [comment]
@RishabhJain2018 make a new PR for challenge details. Let's merge this for now. cc @deshraj

File: frontend/src/js/route-config/route-config.js [hunk 8]
```diff
@@ -183,6 +183,15 @@
             authenticate: true
         };
 
+        var all_challenges = {
+            name: 'all-challenges',
+            url: "/challenges/list",
+            templateUrl: baseUrl + "/web/challenge/list_all_challenges.html",
+            controller: 'ChallengeListAllCtrl',
```
Comment (aka-jain) [inline frontend/src/js/route-config/route-config.js]: Use 'challengeListCtrl' here

**Comment by deshraj** [comment]
@aka-jain feel free to merge this if this is working fine on your system and passing all the checks.

**Comment by aka-jain** [comment]
On it @deshraj

**Comment by deshraj** [comment]
@aka-jain is this PR mergable? Please merge it if ready. Thanks!

File: frontend/src/views/web/landing.html [hunk 9]
```diff
@@ -47,7 +47,7 @@ <h4>
                             <p>
                                 Explore other past, ongoing and upcoming challenges.
                             </p>
-                            <a class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-light fs-14 mt-12" ui-sref="web.challenge-main">View All</a>
+                            <a class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-light fs-14 mt-12" ui-sref="all-challenges">View All</a>
```
Comment (aka-jain) [inline frontend/src/views/web/landing.html]: One more additon, if user is logged in then redirect to 'web.challenge-main' othersiw 'all-challeges'. @xamfy

**Comment by aka-jain** [comment]
@xamfy please do the required changes, then will merge this PR. Thanks.

**Comment by aka-jain** [comment]
@xamfy you working on this. Please let us know asap.

**Comment by aka-jain** [comment]
@RishabhJain2018 can you please take this PR up?

File: frontend/src/views/web/landing.html (hunk 9 above)
Comment (aka-jain) [reply frontend/src/views/web/landing.html]: @RishabhJain2018 can you please do this?

Comment (aka-jain) [reply frontend/src/views/web/landing.html]: I'm taking this @RishabhJain2018

**Comment by aka-jain** [comment]
@spyshiv please review this PR.

**Comment by RishabhJain2018** [comment]
@aka-jain @deshraj @spyshiv I am taking up this feature. :-)

File: frontend/src/js/route-config/route-config.js (hunk 8 above)
Comment (RishabhJain2018) [reply frontend/src/js/route-config/route-config.js]: @aka-jain I think we can't use it here as it also sends a parameter of `user token` in it but we don't have the user token.

**Comment by deshraj** [comment]
Sounds good.

**Pull request closed without merging**